### DISCUSSION TITLE
About identifying binary.

### DISCUSSION THREAD

**Discussion by anzosasuke** [NationalSecurityAgency/ghidra#3098] [General]
I saw when I was using loading c or c++ binary in ghidra,  it knew which dll that is needed. 
Can we know from that .dll library, which language compiled the binary? 
Can ghidra identify which language or compiler the binary is compiled from? I mean, does it have that capability? 

Any thoughts on this would be so much helpful.

**Comment by ryanmkurtz** [comment]
Near the start of the PE headers you'll find an `IMAGE_RICH_HEADER` structure.  If you expand it, you'll find some of the info you desire:

![image](https://user-images.githubusercontent.com/5378554/120633861-a0356a80-c438-11eb-9a54-9dc34c0f95e7.png)

**Reply by anzosasuke** [reply]
I was not able to find image rich header in some binaries of C, i don't know what happened, but thanks for pointing it out.

**Reply by hippietrail** [reply]
I believe the Image Rich Header is optional. I think MSVC adds one and most other compilers don't. I'm not sure if removing it can affect running MSVC code, probably not. It might affect debugging perhaps. But there are probably tools to remove it if it doesn't affect running.
